session_test.go

```diff
@@ -405,19 +405,17 @@ func TestNegotiateStreamError(t *testing.T) {
 	defer cancel()
 	clientConn, serverConn := net.Pipe()
 	clientJID := jid.MustParse("me@example.net")
+	semaphore := make(chan struct{})
 	go func() {
-		s, err := xmpp.ReceiveSession(ctx, serverConn, 0, xmpp.NewNegotiator(xmpp.StreamConfig{
+		_, err := xmpp.ReceiveSession(ctx, serverConn, 0, xmpp.NewNegotiator(xmpp.StreamConfig{
 			Features: func(*xmpp.Session, ...xmpp.StreamFeature) []xmpp.StreamFeature {
 				return []xmpp.StreamFeature{errorStartTLS(stream.Conflict)}
 			},
 		}))
 		if err != nil {
 			t.Logf("error receiving session: %v", err)
 		}
-		err = s.Close()
-		if err != nil {
-			t.Logf("error closing session: %v", err)
-		}
+		semaphore <- struct{}{}
```

Comment:
this is a super minor nitpick, but if we have to make more changes anyways let's just close the channel here instead of sending to it.